File: interviews/leetcode/prefx-sum/1732. Find the Highest Altitude/solution.go
func largestAltitude(gain []int) int {
	maxAltitude := 0
	currentAltitude := 0

	for _, g := range gain {
		currentAltitude += g
		maxAltitude = max(maxAltitude, currentAltitude)
	}

	return maxAltitude
}

/*
func main() {
    gain := []int{-5, 1, 5, 0, -7}
    fmt.Println(largestAltitude(gain)) // Output: 1
}
*/

File: interviews/leetcode/prefx-sum/1732. Find the Highest Altitude/solution.py
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        # compute the max altitude
        max_altitude = 0
        current_altitude = 0

        # cummulate and then take max
        for g in gain:
            current_altitude += g
            max_altitude = max(max_altitude, current_altitude)

        return max_altitude
